RecoverCRPVM/Common-Functions.psm1

```diff
@@ -0,0 +1,408 @@
+    {
+        Write-log "VM ==> $($vm.Name) is not a Windows VM, and is currently not supported by this script, cannot continue exiting." -color Red
+        return $false
+    } #>   
+    return $true
+}
+
+function CreateRescueVM(
+    [Object[]]$vm,
+    [Parameter(mandatory=$true)]
+    [String]$ResourceGroup,
+    [Parameter(mandatory=$true)]
+    [String]$rescueVMNname,
+    [Parameter(mandatory=$true)]
+    [String]$RescueResourceGroup,
+    [String]$prefix = "rescue",
+    [Parameter(mandatory=$false)]
+    [String]$Sku,
+    [Parameter(mandatory=$false)]
+    [String]$Offer,
+    [Parameter(mandatory=$false)]
+    [String]$Publisher,
+    [Parameter(mandatory=$false)]
+    [String]$Version
+    )
+{
+
+    Try
+    {
+        write-log "Initiating the process to create the new Rescue VM" -color Yellow
+        
+        if ($vm.StorageProfile.OsDisk.ManagedDisk) {$managedVM = $true} else  {$managedVM = $false}
+
+        $osDiskName  = $vm.StorageProfile.OsDisk.Name
+        $vmSize = $vm.HardwareProfile.VmSize
+        $osType = $vm.StorageProfile.OsDisk.OsType
+        $location = $vm.Location
+        $networkInterfaceName = $vm.NetworkProfile.NetworkInterfaces[0].Id.split('/')[-1]
+        $MaxStaorageAccountNameLength=24
+
+        $rescueOSDiskName = "$prefix$osDiskName"
+        if (-not $managedVM)
+        {
+            $osDiskVhdUri = $vm.StorageProfile.OsDisk.Vhd.Uri
+            $storageAccountName = $vm.StorageProfile.OsDisk.Vhd.Uri.Split('//')[2].Split('.')[0]
+            $rescueosDiskVhduri = $osDiskVhdUri.Replace($osDiskName,$rescueOSDiskName)
+        }
+
+        $rescuevm = New-AzureRmVMConfig -VMName $rescueVMNname -VMSize $vmSize;
+        $rescuenetworkInterfaceName = "$prefix$networkInterfaceName"
+        $nic1 = Get-AzureRmNetworkInterface   -ResourceGroupName $ResourceGroup | Where-Object {$_.Name -eq $networkInterfaceName}
+        $nic1Id = $nic1.Id
+        $rescuenic1Id = $nic1Id.Replace($networkInterfaceName,$rescuenetworkInterfaceName)
+        $rescuevm = Add-AzureRmVMNetworkInterface -VM $rescuevm -Id $rescuenic1Id
+        $rescuevm.NetworkProfile.NetworkInterfaces[0].Primary = $true
+        #$rescuevm = Set-AzureRmVMOSDisk -VM $rescuevm -VhdUri $rescueosDiskVhduri -name $rescueOSDiskName -CreateOption attach -Windows              
+        $rescueStorageType = "Standard_GRS"
+        $rescueStorageName = "$prefix$storageAccountName"
+        $rescueStorageName = $rescueStorageName.ToLower()
+        if ($rescueStorageName.Length -gt $MaxStaorageAccountNameLength) #ensures that the storage account name is less than 24 characters
+        {
+          $rescueStorageName = $rescueStorageName.Substring(0,$MaxStaorageAccountNameLength)
+        }
+        ## Network
+        $rescueInterfaceName = $prefix+"interface"
+        $rescueSubnet1Name = $prefix + "Subnet"
+        $rescueVNetName = $prefix +"VNet"
+        $rescueVNetAddressPrefix = "10.0.0.0/16"
```

Comment:
Will these values always remain constant, or does these make sense to take as parameters?